### PR TITLE
[K32W0] Add platform changes and features

### PR DESCRIPTION
**Pull request by andrei-menzopol**
This PR adds SVE2 fixes and other changes and features:

* Add support for custom certificate policy.
* Certificate declaration: set certificate_type to 1
* Add new demo certificates
* Add demo certificates allowing testing with two different DUTs

* Add custom lighting app ZAP file - e.g.: removed generic and not needed clusters like the ones required for Ethernet/WiFi; removed the optional clusters for an On/Off Device Type;

* Add ksdk_mbedlts.c such that we redirect all the AES calls through SecLib.c

* Add TestEventTrigger Support

* Add functionality for ResetWatermarks()

* Add callback for PDM logs

* Fix NotifyUpdateApplied - message that is sent when booting for the first time with an OTA image was not being send due to a bug in the PDM read function: ReadConfigValue(...);
* Add delay when rebooting after OTA finishes
* Improve OTA mechanism

* Remove third parameter of ReadConfigValue() and use a TValue& parameter which can be used for extracting the value length

* Consume faster the RX buffer
* RX buffer should be processed with high priority because:
  * 15.4 packets are consumed on Thread task;
  * Thread task has a higher priority than the Matter one (3 > 2)

* Fix Low Power
  * fix 15.4 warm-up time: before this commit calibration time also included the time required for attaching to the Thread network This led to a large warm-up value that blocked some 15.4 operations Fix this by including in the warm-up time only the 15.4 radio initialization time
  * fix the warm-up initial value to avoid any conflicts
  * fix possible overflow in reading timestamp values
  * Make sure that all the logs are disabled when an application is compiled with low-power support; logging can interference with LP timings and break functionality;

* Fix time support for Matter
  * using FreeRTOS timers (based on FreeRTOS tickets) is inaccurate and leads to time drifting
  * use OT Timers API  (Wtimer-based) for getting us/ms timestamps

* Fix ExtractPubkeyFromX509Cert in CHIPCryptoPALTinyCrypt.cpp

* Fix BLE advertising interval + issue warning if the peer doesn't support 2Mbps

* Fix Diagnostic Data Provider

* Fix OnOff attribute persistence over hw reset

* Fix compilation issue for lock-app

* Disable by default on lighting-app BLE Set Phy Req to 2M to be extra cautious during interops

* Update readme files

Signed-off-by: Doru Gucea <doru-cristian.gucea@nxp.com>
Signed-off-by: Marius Tache <marius.tache@nxp.com>
Signed-off-by: Andrei Menzopol <andrei.menzopol@nxp.com>

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
@andrei-menzopol this seems to need fixes for CI to pass.

**Comment by andrei-menzopol** [comment]
@andy31415 Seems to be an issue with restyled when tries to parse a .der certificate. How should I fix this?

File: .github/.wordlist.txt [hunk 1]
```diff
@@ -695,6 +694,7 @@ ipynb
 IPython
 ISCAN
 isHexString
+IsInitiator
```
Comment (andy31415) [inline .github/.wordlist.txt:697]: IsInitiator is not a word

**Comment by andy31415** [comment]
> @andy31415 Seems to be an issue with restyled when tries to parse a .der certificate. How should I fix this?

Maybe adding excludes in .restyled.yaml will work

File: .github/.wordlist.txt (hunk 1 above)
Comment (andrei-menzopol) [reply .github/.wordlist.txt:697]: It believe was previously introduced by other PR, the diff is the result of running `sort -f | unique -i`. I can remove it

File: .github/workflows/examples-k32w.yaml [hunk 2]
```diff
@@ -34,7 +34,7 @@ jobs:
         if: github.actor != 'restyled-io[bot]'
 
         container:
-            image: connectedhomeip/chip-build-k32w:0.5.99
+            image: connectedhomeip/chip-build-k32w:0.6.02
```
Comment (andy31415) [inline .github/workflows/examples-k32w.yaml:37]: If this is required, yaml in integrations/cloudbuild/ have to be updated as well (the build-all and smoke test)

**Pull request merged**